mitiq/cdr/clifford_training_data.py

```diff
@@ -0,0 +1,488 @@
+    """
+    additional_options = additional_options.get("additional_options")
+    # testing the dictionary exists:
+    if bool(additional_options):
+        sigma_select = additional_options.setdefault("sigma_select", 0.5)
+        sigma_replace = additional_options.setdefault("sigma_replace", 0.5)
+    else:
+        additional_options = {}
+        sigma_select = additional_options.setdefault("sigma_select", 0.5)
+        sigma_replace = additional_options.setdefault("sigma_replace", 0.5)
+    sigma_select = additional_options.get("sigma_select")
+    sigma_replace = additional_options.get("sigma_replace")
+    # set the seed for sampling, for replacement and selection:
+    (random_state_select, random_state_replace) = random_state.randint(
+        10 ** (8), size=2
+    )
+    random_state_select = np.random.RandomState(random_state_select)
+    random_state_replace = np.random.RandomState(random_state_replace)
+    # Choose non Clifford gates to change according to selection methods:
+    columns_to_change = _select(
+        rz_non_cliff,
+        fraction_non_clifford,
+        method_select,
+        sigma_select,
+        random_state_select,
+    )
+    rz_non_cliff_selected = rz_non_cliff[columns_to_change]
+    pos_selected = pos_non_cliff[columns_to_change]
+    qubits_selected = qubits_non_cliff[columns_to_change]
+    # Now the non Clifford gates have been selected, we need to decide which
+    # Clifford gate to replace them with.
+    rz_non_cliff_replaced = _replace(
+        rz_non_cliff_selected,
+        method_replace,
+        sigma_select,
+        sigma_replace,
+        random_state_replace,
+    )
+    new_gates = [cirq.ops.rz(parameter) for parameter in rz_non_cliff_replaced]
+    # rebulding the operations:
+    new_operations = []
+    for i, gate in enumerate(new_gates):
+        qubit = qubits_selected[i]
+        new_operations.append(gate(qubit))
+    operations[[int(i) for i in pos_selected]] = new_operations
+    projected_circuit = Circuit(operations)
+    return projected_circuit
+
+
+def _get_arguments(operation: cirq.ops.GateOperation) -> float:
+    """Takes a cirq GateOperation object and returns the exponent multiplied
+    by pi. This corresponds to the angle of a rotation gate.
+
+    Args:
+        operation: a cirq GateOperation.
+
+    Returns:
+        operation.gate.exponent*pi: cirq.ops.GateOperation.gate.exponent*pi"""
+    return operation.gate.exponent * np.pi
+
+
+_get_arguments = np.vectorize(_get_arguments)
+
+
+def _select(
+    rz_non_cliff: np.ndarray,
+    fraction_non_clifford: float,
+    method_select: str,
+    sigma_select: float,
+    random_state: np.random.RandomState,
+) -> np.ndarray:
+    """Function to select the non-Clifford gates to be replace for a given set
+    of  non-Clifford gates.
+
+    Args:
+        rz_non_cliff: array of non-Clifford angles for a circuit of interest.
+        fration_non_clifford: fraction of non-Clifford gates to change.
```

Comment:
```suggestion
        rz_non_clifford: array of non-Clifford angles for a circuit of interest.
        fration_non_clifford: fraction of non-Clifford gates to change.
```

Reply:
Here and elsewhere I would try to be consistent and keep the same kind of abbreviations. Also, it sounds more pythonic to have full word spellings.